src/components/typeahead/typeahead.component.ts

```diff
@@ -221,6 +217,20 @@ export class TypeaheadComponent<T = any> implements OnChanges, OnDestroy {
             }
         }
 
+        // Cut off recentOptions at recentOptionsMaxCount
+        if (changes.recentOptions || changes.recentOptionsMaxCount) {
+            this._recentOptions = this.recentOptions
+                ? this.recentOptions.slice(0, this.recentOptionsMaxCount)
+                : undefined;
+
+            if (changes.recentOptionsMaxCount) {
+                // Avoid ExpressionChangedAfterChecked error
+                setTimeout(() => {

```

Comment:
Event emitters can be asynchronous which should prevent the expression changed error without us having to use timeouts etc.

You could try:

```typescript
@Ouput() recentOptionsChange = new EventEmitter<ReadonlyArray<T>>(true);
```

And that should hopefully negate the need for the timeout

Reply:
I don't really know why, because isAsync seems to do basically the same thing, but changing this breaks the karma tests that I had added.

Reply:
Ok, no worries. Then leave it as is, thanks!